File: RPC10/C.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, m;
    cin >> n >> m;

    vector<bool> inJava(n, false);
    vector<bool> inCpp(n, false);
    for (int i = 0; i < m; i++)
    {
        int p, L;
        cin >> p >> L;
        if (L == 1)
            inCpp[p - 1] = 1;
        if (L == 2)
            inJava[p - 1] = 1;
    }

    for (int i = 0; i < n; i++)
        if (inCpp[i] && inJava[i])
            cout << i + 1 << endl;
    return 0;
}

File: RPC10/C.py
def main():
    n, m = map(int, input().split())

    in_java = [0] * n
    in_cpp = [0] * n

    for _ in range(m):
        p = int(input())
        L = int(input())
        if L == 1:
            in_cpp[p - 1] = 1
        if L == 2:
            in_java[p - 1] = 1

    for i in range(n):
        if in_cpp[i] and in_java[i]:
            print(i + 1)


if __name__ == "__main__":
    main()
